src/material/schematics/ng-generate/theming-api/migration.ts

```diff
@@ -0,0 +1,325 @@
+  // This one was split up into two mixins which is trickier to
+  // migrate so for now we forward to the deprecated variant.
+  'cdk-text-field': 'text-field',
+};
+
+/**
+ * Migrates the content of a file to the new theming API. Note that this migration is using plain
+ * string manipulation, rather than the AST from PostCSS and the schematics string manipulation
+ * APIs, because it allows us to run it inside g3 and to avoid introducing new dependencies.
+ * @param content Content of the file.
+ * @param oldMaterialPrefix Prefix with which the old Material imports should start.
+ *   Has to end with a slash. E.g. if `@import '~@angular/material/theming'` should be
+ *   matched, the prefix would be `~@angular/material/`.
+ * @param oldCdkPrefix Prefix with which the old CDK imports should start.
+ *   Has to end with a slash. E.g. if `@import '~@angular/cdk/overlay'` should be
+ *   matched, the prefix would be `~@angular/cdk/`.
+ * @param newMaterialImportPath New import to the Material theming API (e.g. `~@angular/material`).
+ * @param newCdkImportPath New import to the CDK Sass APIs (e.g. `~@angular/cdk`).
+ */
+export function migrateFileContent(content: string,
+                                   oldMaterialPrefix: string,
+                                   oldCdkPrefix: string,
+                                   newMaterialImportPath: string,
+                                   newCdkImportPath: string): string {
+  // Drop the CDK imports and detect their namespaces.
+  const cdkResults = detectAndDropImports(content, oldCdkPrefix);
+  content = cdkResults.content;
+
+  // Drop the Material imports and detect their namespaces.
+  const materialResults = detectAndDropImports(content, oldMaterialPrefix);
+  content = materialResults.content;
+
+  // If nothing has changed, then the file doesn't import the Material theming API.
+  if (materialResults.hasChanged || cdkResults.hasChanged) {
+    // Replacing the imports may have resulted in leading whitespace.
+    content = content.replace(/^\s+/, '');
+    content = migrateCdkSymbols(content, newCdkImportPath, cdkResults.namespaces);
+    content = migrateMaterialSymbols(content, newMaterialImportPath, materialResults.namespaces);
+  }
+
+  return content;
+}
+
+/**
+ * Finds all of the imports matching a prefix, removes them from
+ * the content string and returns some information about them.
+ * @param content Content from which to remove the imports.
+ * @param prefix Prefix that the imports should start with.
+ */
+function detectAndDropImports(content: string, prefix: string):
+  {content: string, hasChanged: boolean, namespaces: string[]} {
+  if (prefix[prefix.length - 1] !== '/') {
+    // Some of the logic further down makes assumptions about the import depth.
+    throw Error(`Prefix "${prefix}" has to end in a slash.`);
+  }
+
+  const pattern = new RegExp(`@(import|use) +['"]${escapeRegExp(prefix)}.*['"].*;?\n`, 'g');
+  const namespaces: string[] = [];
```

Comment:
Maybe add a comment like
```ts
// List of `@use` namespaces from which Angular CDK/Material APIs may be referenced.
// Since we know that the library doesn't have any name collisions, we can treat all of these
// namespaces as equivalent.
```